lib/stateful_enum/machine.rb

```diff
@@ -18,6 +18,13 @@ def initialize(model, column, states, prefix, suffix, &block)
         @model.send :undef_method, "#{@prefix}#{state}#{@suffix}!"
       end
 
+      model.validate(if: -> { send("#{column}_changed?") && send("#{column}_was") }) do
```

Comment:
@amatsuda I think this now really feels the rails way :D

In order to accomplish that I had to add this new method `StateInspector#stateful_enum_for` it allows `possible_states` to be filter by the current enum column